packages/block-library/src/gallery/transforms.js

```diff
@@ -33,6 +34,33 @@ const parseShortcodeIds = ( ids ) => {
 	return ids.split( ',' ).map( ( id ) => parseInt( id, 10 ) );
 };
 
+function transformV1FormatFromThirdPartyBlocks( block ) {
+	const settings = select( blockEditorStore ).getSettings();
+	if (
+		settings.__unstableGalleryWithImageBlocks &&
+		block.name === 'core/gallery' &&
+		block.attributes?.images
+	) {
+		const innerBlocks = block.attributes.images.map( ( image ) => {
+			return createBlock( 'core/image', {
+				url: image.url,
+				id: image.id,
+			} );
+		} );
```

Comment:
Feel free to ignore. Can be also shortened to:

```js
		const innerBlocks = block.attributes.images.map( ( { url, id } ) =>
			createBlock( 'core/image', {
				url,
				id,
			} )
		);
```

Reply:
How did I overlook that destructuring goodness! Updated.